src/expire.c

```diff
@@ -149,7 +159,26 @@ void expireScanCallback(void *privdata, void *entry) {
     data->sampled++;
 }
 
-static inline int isExpiryTableValidForSamplingCb(hashtable *ht) {
+/* Expires up to `max_entries` fields from a hash with volatile fields.
+ * Sets `has_more_expired_entries` if more remain. Updates stats. */
+void fieldExpireScanCallback(void *privdata, void *volaKey) {
+    expireScanData *data = privdata;
```

Comment:
For safety/readability, can we convert the `void*` parameter into a typed pointer?

```suggestion
void fieldExpireScanCallback(void *privdata, void *volaKeyVoid) {
    expireScanData *data = privdata;
    robj *volaKey = volaKeyVoid;
```